scheduler/plugin/plugin_test.go

```diff
@@ -970,6 +972,100 @@ wasm stack trace:
 	}
 }
 
+func TestReserve(t *testing.T) {
+	tests := []struct {
+		name                   string
+		guestURL               string
+		args                   []string
+		globals                map[string]int32
+		pod                    *v1.Pod
+		nodeName               string
+		expectedPanic          string
+		expectedStatusCode     framework.Code
+		expectedStatusMessage  string
+		expectedUnreserveError string
+	}{
+		{
+			name:               "Success",
+			pod:                test.PodSmall,
+			nodeName:           "good",
+			expectedStatusCode: framework.Success,
+		},
+		{
+			name:                  "Error",
+			pod:                   test.PodSmall,
+			nodeName:              "bad",
+			expectedStatusCode:    framework.Error,
+			expectedStatusMessage: "name is bad",
+		},
+		{
+			name:     "reachable: flag is 0",
+			guestURL: test.URLTestReserveFromGlobal,
+			pod:      test.PodSmall,
+			nodeName: test.NodeSmall.Name,
+			globals:  map[string]int32{"flag": 0},
+		},
+		{
+			name:     "unreachable: flag is 1",
+			guestURL: test.URLTestReserveFromGlobal,
+			pod:      test.PodSmall,
+			// expectedStatusCode: framework.Error,
+			nodeName:               test.NodeSmall.Name,
+			globals:                map[string]int32{"flag": 1},
+			expectedUnreserveError: "wasm: unreserve error: wasm error: unreachable\nwasm stack trace:\n\treserve_from_global.$1()",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			guestURL := tc.guestURL
+			if guestURL == "" {
+				guestURL = test.URLTestReseve
+			}
+
+			p, err := wasm.NewFromConfig(ctx, "wasm", wasm.PluginConfig{GuestURL: guestURL, Args: tc.args})
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer p.(io.Closer).Close()
+
+			if len(tc.globals) > 0 {
+				pl := wasm.NewTestWasmPlugin(p)
+				pl.SetGlobals(tc.globals)
+			}
+
+			defer func() {
```

Comment:
nit: you don't use expectedPanic, so you can remove it :)
BTW I guess you don't need to catch panic because status should have panic's message

Reply:
605dc6f